replica/replica.go

```diff
@@ -606,37 +612,19 @@ func (r *Replica) processPrepareRemoveDisks(disks []string) ([]PrepareRemoveActi
 			return nil, fmt.Errorf("Wrong disk %v doesn't exist", disk)
 		}
 
-		children := r.diskChildrenMap[disk]
-		// 1) leaf node
-		if children == nil {
-			actions = append(actions, PrepareRemoveAction{
+		parent := r.diskData[disk].Parent
+
+		actions = append(actions,
+			PrepareRemoveAction{
+				Action: OpCoalesce,
+				Source: disk,
+				Target: parent,
+			},
+			PrepareRemoveAction{
 				Action: OpRemove,
```

Comment:
there are checks for count of the children in removeDiskNode to error out if children count is > 1.
would that fine or should be handled?

Reply:
handled